### PR TITLE
Add Notification package

### PR DESCRIPTION
**Pull request by hharnisc**
### Purpose

Display in app notifications!

### Notes

notifications

### Review

Let's replicate this animation technique for posts! 

#### Staging Deployment

_This repo is CI/CD enabled and staging deployment should be available at:
https://task-add-notification-package.publish.buffer.com :smile:_

### CODE REVIEW

**Comment by hharnisc** [comment]
Merging this one in :green_apple:

File: packages/notifications/middleware.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import {
+  transitionTimeMilliseconds,
+} from '@bufferapp/components/style/animation';
+import {
+  actionTypes,
+  actions,
+} from './reducer';
+
+export default ({ dispatch }) => next => (action) => { // eslint-disable-line no-unused-vars
```
Comment (hamstu) [inline packages/notifications/middleware.js]: Thinking you probably don't need the `// eslint-disable-line no-unused-vars` anymore?

File: packages/notifications/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# @bufferapp/notifications
```
Comment (hamstu) [inline packages/notifications/README.md]: Love the documentation Harrison! 😍

File: packages/notifications/components/Notifications/test-harness.story.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    notificationType: 'error',
+    message: 'Oh No! There was an error doing the thing!',
+  },
+  {
+    notificationType: 'success',
+    message: 'Success! Full cheese has been activated.',
+  },
+  {
+    notificationType: 'error',
+    message: 'Oops! Cheese was dropped',
```
Comment (hamstu) [inline packages/notifications/components/Notifications/test-harness.story.jsx]: 🧀 😱

File: packages/notifications/middleware.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      setTimeout(() => {
+        dispatch(actions.createNotificationComplete({
+          notification,
+        }));
+      });
+      setTimeout(() => {
+        dispatch(actions.deleteNotification({
+          notification,
+        }));
+      }, 5000);
```
Comment (hamstu) [inline packages/notifications/middleware.js]: Should we define this `5000` ms in a file somewhere?

File: packages/notifications/middleware.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+          notification: action.notification,
+        })), transitionTimeMilliseconds);
+      break;
+    case actionTypes.CREATE_NOTIFICATION: {
+      const { type, ...notification } = action;
+      setTimeout(() => {
+        dispatch(actions.createNotificationComplete({
+          notification,
+        }));
+      });
```
Comment (hamstu) [inline packages/notifications/middleware.js]: Do we need the `setTimeout` with no timeout value here? Not sure if it's to ensure a render/state change happens first. 🤔

**Pull request merged**